Dockerfile

```diff
@@ -0,0 +1,116 @@
+# syntax=docker/dockerfile:1
```

Comment:
When the average user downloads project and open this file to understand what he would build as docker image, it would be quite confusing to him. Previous docker files for `worker` and `leader` had duplication, but were very simple and obvious. 
If the rest of the team wants to go this way for docker builds, then I am fine with it, but I don't like it very much.